ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -542,6 +543,10 @@ public void setUnderlineColor(ReactEditText view, @Nullable Integer underlineCol
       }
     }
 
+    if (Build.VERSION.SDK_INT == Build.VERSION_CODES.LOLLIPOP) { 
+      setBorderColor(view, 4, Color.TRANSPARENT);
+    }
+
     if (underlineColor == null) {
       drawableToMutate.clearColorFilter();
```

Comment:
the borderColor should return to default color if prop underlineColorAndroid is not passed

Reply:
default `underlineColorAndroid` is `transparent`

https://github.com/facebook/react-native/blob/704dd2812f7b8c79971274cc9e4c717e56847ac0/Libraries/Components/TextInput/TextInput.js#L1170